services/cd-service/pkg/repository/transformer_test.go

```diff
@@ -39,6 +39,98 @@ const (
 	additionalVersions = 7
 )
 
+// Tests various error cases in the Undeploy endpoint, specifically the error messages returned.
+func TestUndeployErrors(t *testing.T) {
+	tcs := []struct {
+		Name         string
+		Transformers []Transformer
+		expectedError string
+		expectedCommitMsg string
+		shouldSucceed bool
+	}{
+		{
+			Name:         "Access non-existent application",
+			Transformers:  []Transformer{
+				&CreateUndeployApplicationVersion{
+					Application: "app1",
+				},
+			},
+			expectedError: "cannot undeploy non-existing application 'app1'",
+			expectedCommitMsg: "",
+			shouldSucceed: false,
+		},
+		{
+			Name:         "Success",
+			Transformers:  []Transformer{
+				&CreateApplicationVersion{
+					Application: "app1",
+					Manifests: map[string]string{
+						envProduction: "productionmanifest",
+					},
+				},
+				&CreateUndeployApplicationVersion{
+					Application: "app1",
+				},
+			},
```

Comment:
what should happen if we create a new version after the UndeployVersion?

Reply:
I will add a test for this

Reply:
It should "just work", so the new version would "win". I added a test